src/gafaelfawr/models/token.py

```diff
@@ -124,7 +124,9 @@ class TokenGroup(BaseModel):
         regex=GROUPNAME_REGEX,
     )
 
-    id: int = Field(..., title="The numeric GID of the group", example=123181)
+    id: Optional[int] = Field(
```

Comment:
Yeah, I'm not really happy with this whole design, since the model as returned to the user will always have that field populated and the only reason why it's optional is because the same model is being used internally to pass data around. I need to do the refactoring work to split it into two models, and have the public model not be `Optional`. I've added that to my to-do list.